pkg/plugins/policies/matchers/match.go

```diff
@@ -1,14 +1,104 @@
 package matchers
 
 import (
+	"sort"
+
+	"github.com/pkg/errors"
+
+	common_proto "github.com/kumahq/kuma/api/common/v1alpha1"
+	mesh_proto "github.com/kumahq/kuma/api/mesh/v1alpha1"
 	core_mesh "github.com/kumahq/kuma/pkg/core/resources/apis/mesh"
-	"github.com/kumahq/kuma/pkg/core/resources/model"
+	core_model "github.com/kumahq/kuma/pkg/core/resources/model"
 	core_xds "github.com/kumahq/kuma/pkg/core/xds"
 	xds_context "github.com/kumahq/kuma/pkg/xds/context"
 )
 
+type ResourceSpecWithTargetRef interface {
+	core_model.ResourceSpec
+	GetTargetRef() *common_proto.TargetRef
+}
+
 // MatchedPolicies match policies using the standard matchers using targetRef (madr-005)
-func MatchedPolicies(model.ResourceType, *core_mesh.DataplaneResource, xds_context.Resources) (core_xds.TypedMatchingPolicies, error) {
-	// TODO @lobkovilya Implement standard matching strategy
-	return core_xds.TypedMatchingPolicies{}, nil
+func MatchedPolicies(rType core_model.ResourceType, dpp *core_mesh.DataplaneResource, resources xds_context.Resources) (core_xds.TypedMatchingPolicies, error) {
```

Comment:
imho it's a bit weird to see this in `plugins` package. This is the core logic of Kuma.

Reply:
But will you use this code outside of the `plugins/polcieis`?

Reply:
yeah, ok. We can always move it in the future